go-controller/pkg/ovn/obj_retry.go

```diff
@@ -126,18 +189,1067 @@ func (r *retryObjs) getObjRetryEntry(key string) *retryObjEntry {
+	for {
+		select {
+		case <-time.After(retryObjInterval):
+			klog.V(5).Infof("%s s have elapsed, retrying failed objects of type %v", retryObjInterval, r.oType)
+			oc.iterateRetryResources(r, false)
+
+		case <-r.retryChan:
+			klog.V(5).Infof("Retry channel got triggered: retrying failed objects of type %v", r.oType)
+			oc.iterateRetryResources(r, true)
+
+		case <-oc.stopChan:
+			klog.V(5).Infof("Stop channel got triggered: will stop retrying failed objects of type %v", r.oType)
+			return
+		}
+	}
+}
+
+// Given a *retryObjs instance, getSyncResourcesFunc retuns the sync function for a given resource type.
+// This will be then called on all existing objects when a watcher is started.
+func (oc *Controller) getSyncResourcesFunc(r *retryObjs) (func([]interface{}), error) {
+
+	var syncRetriableFunc func([]interface{}) error
+	var syncFunc func([]interface{})
+	var name string
+
+	// If a type needs a retriable sync funcion, it will set syncRetriableFunc
+	// and will keep syncFunc=nil. For a non-retriable sync func,
+	// it will directly set syncFunc.
+	switch r.oType {
+	case factory.PodType:
+		name = "SyncPods"
+		syncRetriableFunc = oc.syncPodsRetriable
+
+	case factory.PolicyType:
+		name = "syncNetworkPolicies"
+		syncRetriableFunc = oc.syncNetworkPolicies
+
+	case factory.NodeType:
+		name = "syncNodes"
+		syncRetriableFunc = oc.syncNodesRetriable
+
+	case factory.PeerServiceTypeVar,
+		factory.PeerNamespaceAndPodSelectorTypeVar:
+		name = ""
+		syncRetriableFunc = nil
+
+	case factory.PeerPodSelectorTypeVar:
+		name = "PeerPodSelector"
+		extraParameters := r.extraParameters.(*NetworkPolicyExtraParameters)
+		syncRetriableFunc = func(objs []interface{}) error {
+			return oc.handlePeerPodSelectorAddUpdate(extraParameters.gp, objs...)
+		}
+
+	case factory.PeerPodForNamespaceAndPodSelectorType:
+		name = "PeerPodForNamespaceAndPodSelector"
+		extraParameters := r.extraParameters.(*NetworkPolicyExtraParameters)
+		syncRetriableFunc = func(objs []interface{}) error {
+			return oc.handlePeerPodSelectorAddUpdate(extraParameters.gp, objs...)
+		}
+
+	case factory.PeerNamespaceSelectorTypeVar:
+		name = "PeerNamespaceSelector"
+		extraParameters := r.extraParameters.(*NetworkPolicyExtraParameters)
+		// the function below will never fail, so there's no point in making it retriable...
+		syncFunc = func(i []interface{}) {
+			// This needs to be a write lock because there's no locking around 'gress policies
+			extraParameters.np.Lock()
+			defer extraParameters.np.Unlock()
+			// We load the existing address set into the 'gress policy.
+			// Notice that this will make the AddFunc for this initial
+			// address set a noop.
+			// The ACL must be set explicitly after setting up this handler
+			// for the address set to be considered.
+			extraParameters.gp.addNamespaceAddressSets(i)
+		}
+
+	case factory.LocalPodSelectorTypeVar:
+		name = "LocalPodSelectorTypeVar"
+		syncRetriableFunc = r.syncFunc
```

Comment:
talking with @dcbw I'm not sure we need a sync function for this. This is really about hanldeInitialSelectedPods from network policy:
https://github.com/ovn-org/ovn-kubernetes/blob/6698b01bd6e79bd82ecb9d406904cf6631d72a15/go-controller/pkg/ovn/policy.go#L1073

But I think this is not needed because all the add events will go through the normal onAdd (addResource) path as well. I think @jcaamano added this or reworked it. Jaime do you agree on removing this?

Reply:
I don't think I would unless there is some other reason to remove it. While handleInitialItems is functionally equivalent to the successive OnAdd that would be called after, it is more efficient as it handles all initial items on one shot and accounts for all of them on the initial creation of the policy nbdb PGs, whereas successive OnAdd would handle them on successive updates towards nbdb.

Reply:
ok so its so that we execute potentially less txns/operations, let's just leave it for now